src/odemis/driver/test/tescan_test.py

```diff
@@ -133,6 +145,177 @@ def test_pickle(self):
+
+        for child in cls.sem.children.value:
+            if child.name == CONFIG_SED["name"]:
+                cls.sed = child
+            elif child.name == CONFIG_SCANNER["name"]:
+                cls.scanner = child
+            elif child.name == CONFIG_STG["name"]:
+                cls.stage = child
+            elif child.name == CONFIG_FOCUS["name"]:
+                cls.focus = child
+
+    @classmethod
+    def tearDownClass(cls):
+        if TEST_NOHW:
+            return
+
+        cls.sem.terminate()
+        time.sleep(3)
+
+    def setUp(self):
+        if TEST_NOHW:
+            self.skipTest("No hardware present")
+
+    def tearDown(self):
+        pass
+
+    def test_move_abs(self):
+        """
+        Test if it's possible to move the stage in absolute coordinates
+        In addition also test the edge of the range of the rz axis
+        """
+        start_pos = self.stage.position.value.copy()
+        new_pos = {"x": start_pos["x"] + 5e-4, "y": start_pos["y"] + 10e-4, "z": start_pos["z"] + 12e-5,
+                   "rx": start_pos["rx"] - math.radians(2.5), "rz": start_pos["rz"] + math.radians(3.5)}
+
+        f = self.stage.moveAbs(new_pos)
+        f.result()
+        testing.assert_pos_almost_equal(self.stage.position.value, new_pos)
+
+        # move exactly on the edge of the max range of the rz axis
+        new_pos["rz"] = math.radians(360)
+        f = self.stage.moveAbs(new_pos)
+        f.result()
+        new_pos["rz"] = 0.0
+        testing.assert_pos_almost_equal(self.stage.position.value, new_pos)
+
+        # go back to the original starting point
+        self.stage.moveAbsSync(start_pos)
+
+    def test_move_rel(self):
+        """
+        Test if it's possible to move the stage with relative moves with regular move requests
+        """
+        start_pos = self.stage.position.value.copy()
+
+        # test some of the linear axes and especially the newly added rotational axes
+        f = self.stage.moveRel({"x": 2e-6, "y": 3e-6, "rx": math.radians(1), "rz": -math.radians(15)})
+        f.result()
+
+        self.assertNotEqual(self.stage.position.value, start_pos)
+
+        f = self.stage.moveRel({"x": -2e-6, "y": -3e-6, "rx": -math.radians(1), "rz": math.radians(15)})
+        f.result()
+        testing.assert_pos_almost_equal(self.stage.position.value, start_pos, atol=0.1e-6)
+
+        # go back to the original starting point
+        self.stage.moveAbsSync(start_pos)
+        self.assertEqual(self.stage.position.value, start_pos)
+
+    def test_move_out_of_range(self):
+        """
+        test if a relative move and an absolute move out of the range generate ValueErrors
+        """
+        start_pos = self.stage.position.value.copy()
+
+        # Try a relative move outside the range (less than min)
+        axes = self.stage.axes
+        with self.assertRaises(ValueError):
+            self.stage.moveRel({"x": axes["x"].range[1] - axes["x"].range[0] + 10e-6})
```

Comment:
usually the range[0] < range[1], so this would move it to more than the max range, not less than min, which is mentioned in the comment.